contracts/transfer/src/transfer.rs

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use core::convert::TryFrom;
+
+use alloc::vec::Vec;
+use canonical::{Canon, Store};
+use canonical_derive::Canon;
+use dusk_bls12_381::BlsScalar;
+use dusk_bytes::Serializable;
+use dusk_jubjub::JubJubAffine;
+use dusk_kelvin_map::Map;
+use dusk_pki::PublicKey;
+use phoenix_core::{Crossover, Fee, Message, Note};
+
+mod tree;
+use tree::Tree;
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord, Canon)]
```

Comment:
Can't you simply type-alias? Derive all of these things for a new struct is even more code that we add to the `wasm` module.

Reply:
Fixed